### ISSUE TITLE
Button: hierarchies and types are not correctly mapped to antd type and variant

### ISSUE THREAD

**Issue by malta895** [mia-platform/design-system#721]
## A snippet of code for replicating the issue or showing the proposal usage if applicable

![image](https://github.com/user-attachments/assets/f430e09a-f885-4583-a1f3-7aa2ca50c979)
![image](https://github.com/user-attachments/assets/cf36de8f-e391-4af5-b108-f05364e3eb7f)

### Your environment

node:

os:

**Comment by fredmaggiowski** [comment]
Button styling is currently customized in the design system, however with [v5.21.0](https://github.com/ant-design/ant-design/releases/tag/5.21.0) antd introduced button variants that could help us remove customizations and use their own implementation instead

**Comment by malta895** [comment]
> Button styling is currently customized in the design system, however with [v5.21.0](https://github.com/ant-design/ant-design/releases/tag/5.21.0) antd introduced button variants that could help us remove customizations and use their own implementation instead

Yes I saw the variant in the docs, I see from our package.json that we are already on the v5.21.1, so yes I agree with you, we could use variants to customize the button for our needs
